cmd/spruce/main_test.go

```diff
@@ -162,7 +162,8 @@ func TestMain(t *testing.T) {
 			stdout = fmt.Sprintf(format, args...)
 		}
 		var stderr string
-		printfStdErr = func(format string, args ...interface{}) {
+		//Edit log stderr function
+		PrintfStdErr = func(format string, args ...interface{}) {
```

Comment:
The tests validate that what went to stderr is what's expected. In order to test it, we need a hook to capture stderr in our tests. That's the string above. The problem is that we actually have two different spots in the code where printing to stderr is done. Hard errors propagate back up to main, and therefore get printed using the printfStdErr in main, which has been testable. Things that don't stop everything and propagate up to main use the log libraries printf function indirectly (DEBUG, TRACE, and now Warn()... we just weren't testing debug and trace before). 

Both the log and main versions of the function were doing the same thing. It's just that the log one was private and therefore not usable for testing. So I made it public, and rerouted the main one through the log one.

Even cleaner might be to update all the calls to printfstderr in main to use the log library version and get rid of the one in main. Log library is for printing out the logs... why aren't we using it everywhere?